server/service/linux_mdm_test.go

```diff
@@ -0,0 +1,103 @@
+package service
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/mock"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLinuxHostDiskEncryptionStatus(t *testing.T) {
+	ds := new(mock.Store)
+	svc, ctx := newTestService(t, ds, nil, nil)
+
+	actionRequired := fleet.DiskEncryptionActionRequired
+	verified := fleet.DiskEncryptionVerified
+	failed := fleet.DiskEncryptionFailed
+
+	testcases := []struct {
+		name              string
+		host              fleet.Host
+		keyExists         bool
+		clientErrorExists bool
+		status            fleet.HostMDMDiskEncryption
+		notFound          bool
+	}{
+		{
+			name:              "no key",
+			host:              fleet.Host{ID: 1, Platform: "ubuntu"},
+			keyExists:         false,
+			clientErrorExists: false,
+			status: fleet.HostMDMDiskEncryption{
+				Status: &actionRequired,
+			},
```

Comment:
What's the difference between this test and the `"key not found",` one?

Reply:
ah, the difference is if:
1. the the datastore method returns no row
2. the datastore returns a row with an empty key value

Reply:
OK, last question, what are the real world scenarios for each?

I can see:
1. disk is encrypted but not escrowed
2. ?

Reply:
2. device attempted to escrow but hit a client err (stored in `host_disk_encryption_keys.client_error`)